packages/hardhat-core/src/internal/hardhat-network/provider/TxPool.ts

```diff
@@ -48,7 +48,8 @@ export function serializeTransaction(
 
 export function deserializeTransaction(
   tx: SerializedTransaction,
-  common: Common
+  common: Common,
+  { allowUnlimitedContractSize }: { allowUnlimitedContractSize: boolean }
```

Comment:
A lot of the changes in this PR come from the fact that we need to pass the value of `allowUnlimitedContractSize` to be able to create transactions with the correct `disableMaxInitCodeSizeCheck` value.

Reply:
I think this is a validation that is meant to be run only when receiving a new transaction, isn't it?

If that's the case we can run it in the RPC layer / Node, and use `true` everywhere else. WDYT?

Reply:
You mean always using `disableMaxInitCodeSizeCheck: true`, and making the initcode check ourselves in an upper layer? Makes sense to me, I'll do that.